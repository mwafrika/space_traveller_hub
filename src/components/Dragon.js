import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

const Dragon = () => {
  const dispatch = useDispatch();
  const dragon = useSelector((state) => state.dragonReducer);
  useEffect(() => {
    dispatch(dragon());
    console.log('dragons', dragon);
  }, []);

  return (
    <div>
      <h1>My dragons</h1>
      <h1>My dragons</h1>
      <h1>My dragons</h1>
    </div>
  );
};

export default Dragon;
