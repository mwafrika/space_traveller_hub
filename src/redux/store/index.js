/* eslint-disable no-underscore-dangle */
import { routerReducer } from 'react-router-redux';
import {
  combineReducers, createStore, applyMiddleware, compose,
} from 'redux';
import thunk from 'redux-thunk';
import dragonReducer from '../reducers/dragon';
import RocketReducer from '../reducers/rocket';

const rootReducer = combineReducers({
  dragons: dragonReducer,
  rockets: RocketReducer,
  routing: routerReducer,
});

const store = createStore(
  rootReducer,
  { dragons: [] },

  compose(
    applyMiddleware(thunk),
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
  ),
);

export default store;
