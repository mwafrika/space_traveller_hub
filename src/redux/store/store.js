/* eslint-disable no-underscore-dangle */
import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
<<<<<<< HEAD:src/redux/store/index.js
import missions from '../reducers/missions';

const rootReducer = combineReducers({ missions });
=======
import dragonReducer from '../reducers/dragon';
import RocketReducer from '../reducers/rocket';

const rootReducer = combineReducers({
  dragons: dragonReducer,
  rockets: RocketReducer,
});
>>>>>>> develop:src/redux/store/store.js

const store = createStore(rootReducer, applyMiddleware(thunk));

export default store;
