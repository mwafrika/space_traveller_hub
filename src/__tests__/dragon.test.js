/* eslint-disable jsx-quotes */
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/store';
import '@testing-library/jest-dom';
import Dragon from '../components/dragon/dragon';

describe('Test profile page', () => {
  it('should print dragons very well', () => {
    const dragon = {
      id: 'dragon1',
      name: 'Dragon 1',
      flickr_images: 'https://imgur.com/DaCfMsj.jpg',
      type: 'Rocket',
      reserved: false,
    };
    const tree = renderer.create(
      <Provider store={store}>
        <Dragon key={dragon.id} dragon={dragon} />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
