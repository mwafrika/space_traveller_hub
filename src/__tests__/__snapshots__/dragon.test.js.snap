// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test profile page should print dragons very well 1`] = `
<section
  className="dragonCard"
>
  <div
    className="dragonImg"
  >
    <img
      alt="dragon"
      src="https://imgur.com/DaCfMsj.jpg"
    />
  </div>
  <div
    className="dragonText"
  >
    <div
      className="dragonContent"
    >
      <h3>
        Dragon 1
      </h3>
      <h3>
        Rocket
      </h3>
    </div>
    <button
      className="reserveDragon"
      onClick={[Function]}
      type="button"
    >
      Reserve Dragon
    </button>
  </div>
</section>
`;
